app/models/assignment.rb

```diff
@@ -889,16 +895,14 @@ def has_peer_review_assignment?
 
   def create_peer_review_assignment_if_not_exist
     return unless has_peer_review && Assignment.where(parent_assessment_id: id).empty?
-    peerreview_assignment_properties = AssignmentProperties.new
-    peerreview_assignment_properties.token_period = 1
-    peerreview_assignment_properties.non_regenerating_tokens = false
-    peerreview_assignment_properties.unlimited_tokens = false
-    peerreview_assignment_properties.repository_folder = repository_folder
     peerreview_assignment = Assignment.new
     peerreview_assignment.parent_assignment = self
     peerreview_assignment.submission_rule = NoLateSubmissionRule.new
     peerreview_assignment.assignment_stat = AssignmentStat.new
-    peerreview_assignment.assignment_properties = peerreview_assignment_properties
+    peerreview_assignment.assignment_properties.token_period = 1
```

Comment:
Because of the proxying, you should be able to just do `peerreview_assignment.token_period = 1` etc. here and below.